### PR TITLE
Add Zookeeper plugin

### LINKED ISSUE #654: Support the ZooKeeper plugin
Could you please add support for the ZooKeeper plugin?

https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_zookeeper

### PR DESCRIPTION
**Pull request by jamtur01**
Fixes #654.

### CODE REVIEW

File: manifests/plugin/zookeeper.pp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class collectd::plugin::zookeeper (
+  $ensure           = 'present',
```
Comment (bastelfreak) [inline manifests/plugin/zookeeper.pp]: can you add puppet4 datatypes here?

File: templates/plugin/zookeeper.conf.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<Plugin "zookeeper">
```
Comment (bastelfreak) [inline templates/plugin/zookeeper.conf.erb]: could you migrate this to a epp template?

File: spec/classes/collectd_plugin_zookeeper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+require 'spec_helper'
+
+describe 'collectd::plugin::zookeeper', type: :class do
+  let :facts do
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_zookeeper_spec.rb]: can you use rspec-puppet.-facts here?

**Comment by jamtur01** [comment]
I've not migrated anything to Puppet 4 as yet and I haven't spent any time on types or epp. So if someone wants to pick it up that's great otherwise I don't have the bandwidth. Sorry.

**Comment by bastelfreak** [comment]
Hi @jamtur01, thanks for the PR. I did some inline comments. Let me know if you've any questions about them. We recently dropped puppet3 support so it would be cool if all new code uses puppet4 features.

**Comment by bastelfreak** [comment]
@jamtur01 do you need this in a puppet3 environment?

**Comment by jamtur01** [comment]
@bastelfreak Yes but I am happy to use my fork too.

**Comment by bastelfreak** [comment]
Hi @jamtur01  Can you please rebase it against latest master?

**Comment by jamtur01** [comment]
Rebased.

**Comment by bastelfreak** [comment]
@jamtur01 sorry again, but could you rebase a final time? we fixed the spec issues in master

**Comment by jamtur01** [comment]
Rebased again.

**Comment by bastelfreak** [comment]
\o/

File: manifests/plugin/zookeeper.pp [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class collectd::plugin::zookeeper (
+  Enum['present', 'absent'] $ensure           = 'present',
+  Optional[Integer]         $interval         = undef,
+  String                    $zookeeper_host   = 'localhost',
+  String                    $zookeeper_port   = '2181',
```
Comment (bastelfreak) [inline manifests/plugin/zookeeper.pp]: datatypes \o/

File: spec/classes/collectd_plugin_zookeeper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      is_expected.to contain_file('write_riemann.load').with(ensure: 'present')
+      is_expected.to contain_file('write_riemann.load').with(path: '/etc/collectd.d/10-zookeeper.conf')
+      is_expected.to contain_file('write_riemann.load').with(content: %r{Host "myhost"})
+      is_expected.to contain_file('write_riemann.load').with(content: %r{Port "2181"})
+    end
+  end
+
+  context ':ensure => absent' do
+    let :params do
+      { zookeeper_host: ['myhost'], ensure: 'absent' }
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_zookeeper_spec.rb]: How did this pass the tests. This is clearly wrong. Defined datatype is String, but this is a tuple. Strange

File: spec/classes/collectd_plugin_zookeeper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  end
+
+  context ':ensure => present and :zookeeper_host => \'myhost\'' do
+    let :params do
+      { zookeeper_host: 'myhost', zookeeper_port: '2181' }
+    end
+
+    it 'Will create /etc/collectd.d/10-zookeeper.conf' do
+      is_expected.to contain_file('write_riemann.load').with(ensure: 'present')
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_zookeeper_spec.rb]: write_riemann.load here is also wrong. no idea how this went green on travis

**Comment by bastelfreak** [comment]
I fixed the broken tests in https://github.com/voxpupuli/puppet-collectd/pull/698

**Pull request merged**